flystar/motion_model.py

```diff
@@ -0,0 +1,534 @@
+import numpy as np
+from abc import ABC
+import pdb
+from flystar import parallax
+from astropy.time import Time
+from scipy.optimize import curve_fit
+import warnings
+
+class MotionModel(ABC):
+    # Number of data points required to fit model
+    n_pts_req = 0
+    # Degrees of freedom for model
+    n_params = 0
+
+    # Fit paramters: Shared fit parameters
+    fitter_param_names = []
+
+    # Fixed parameters: These are parameters that are required for the model, but are not 
+    # fit quantities. For example, RA and Dec in a parallax model.
+    fixed_param_names = []
+    fixed_meta_data = []
+
+    # Non-fit paramters: Custom paramters that will not be fit.
+    # These parameters should be derived from the fit parameters and
+    # they must exist as a variable on the model object
+    optional_param_names = []
+
+    def __init__(self, *args, **kwargs):
+        # TODO: do we need this?
+        '''for param in self.fitter_param_names:
```

Comment:
This bit of code takes every parameter and sets it as an instance variable. This is very handy later on (we use it in BAGLE fitting and plotting). But up to you whether you want to use it.